### ISSUE TITLE
Add support for Yarn 2

### ISSUE THREAD

**Issue by mnonnenmacher** [heremaps/oss-review-toolkit#2283]
Add support for Yarn 2 without breaking compatibility with Yarn 1. This requires testing if Yarn 2 is fully backward compatible or if we need to have both versions installed in Docker/CI.

For details about Yarn 2 see: https://dev.to/arcanis/introducing-yarn-2-4eh1

**Comment by sschuberth** [comment]
Also see https://www.heise.de/hintergrund/Yarn2-schneller-stabiler-und-moderner-4668679.html (in German).

I wonder if, in a first implementation, we should focus on parsing Yarn-2-lockfiles (which are now valid YAML) *only*, i.e. don't run `yarn install`. This goes in line with the idea we briefly discussed a while ago to separate out lockfile-parsing-based package manager support for package managers that support lockfiles, so we could easily only enable package manager implementations that rely on lock files, for a "quick" run of the analyzer.

**Comment by porsche-rishisaxena** [comment]
@sschuberth: Is there any outcome on adding support for Yarn - 2 in terms of roadmap and timelines?

**Comment by sschuberth** [comment]
No, this as been postponed from our side in favor of looking at https://github.com/oss-review-toolkit/ort/issues/4365. So, we do not currently plan to work on Yarn 2 support.

**Comment by sschuberth** [comment]
Note to myself and @mnonnenmacher: Luckily, Yarn 2+ lock files *do* contain the [dependency hierarchy](https://github.com/yarnpkg/berry/blob/a72fdf6641ed538bf7793a1a0bbc7a88a6bc9be3/yarn.lock#L19).

**Comment by sschuberth** [comment]
Looks like Yarn 3 finally added a way to [create lock files without installing dependencies](https://github.com/yarnpkg/yarn/issues/5738#issuecomment-905943984) via `yarn install --mode update-lockfile`.

**Comment by nnobelis** [comment]
Implemented with https://github.com/oss-review-toolkit/ort/pull/5405.
